api/form.go

```diff
@@ -71,19 +83,23 @@ func PostForm(c httpClient, u string, params url.Values) (*FormResponse, error)
 		requestURI: u,
 	}
 
-	if contentType(resp.Header.Get("Content-Type")) == formType {
-		var bb []byte
-		bb, err = ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return r, err
-		}
+	var bb []byte
+	bb, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return r, err
+	}
 
+	ct := contentType(resp.Header.Get("Content-Type"))
+	r.contentType = ct
+	if ct == formType {
 		r.values, err = url.ParseQuery(string(bb))
 		if err != nil {
 			return r, err
 		}
-	} else {
-		_, err = io.Copy(ioutil.Discard, resp.Body)
+	} else if ct == jsonType {
+		jObj := make(map[string]interface{})
+		err := json.Unmarshal(bb, &jObj)
```

Comment:
37  api / form.go 
@@ -1,8 +1,8 @@
paket api

içe aktar (
	"kodlama / json"
	"fmt"
	"io"
	"io / ioutil"
	"net / http"
	"net / url"
@@ -19,11 +19,23 @@

	requestURI  dizesi
	değerler      url'si. Değerler

	json  haritası [ dize ] arayüzü {}

	contentType  dizesi
}

// k adlı yanıt değerini alın.
func ( f  FormResponse ) Get ( k  string ) string {
	dönüş  f . değerler . Al ( k )
	eğer  f . contentType  ==  formType {
		dönüş  f . değerler . Al ( k )
	} else {
 Api / form.go'da 32. satırdaki hatayı kontrol edin


GitHub Eylemleri
/ lint
api / form.go # L32
`if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
		eğer  v , tamam  : =  f . json [ k ]; tamam {
			dönüş  v . ( dize )
		}


Leave a comment

		 "" döndür
	}
}

// Err, yanıttan çıkarılan bir Error nesnesi döndürür.
@@ -71,19 +83,23 @@
		requestURI : u ,
	}

	if  contentType ( resp . Header . Get ( "Content-Type" )) ==  formType {
		var  bb [] bayt
		bb , err  =  ioutil . ReadAll ( solunum . Vücut )
		eğer  hata  ! =  nil {
			dönüş  r , err
		}
	var  bb [] bayt
	bb , err  =  ioutil . ReadAll ( solunum . Vücut )
	eğer  hata  ! =  nil {
		dönüş  r , err
	}

	ct  : =  contentType ( solunum . Başlık . Get ( "Content-Type" ))
	r . contentType  =  ct
	eğer  ct  ==  formType {
		r . değerler , err  =  url . ParseQuery ( dize ( bb ))
		eğer  hata  ! =  nil {
			dönüş  r , err
		}
	} else {
		_ , err  =  io . Kopya ( ioutil . Sil , solunum . Vücut )
	} else  if  ct  ==  jsonType {
		jObj  : =  make ( harita [ dize ] arayüzü {})
		hata  : =  json . Unmarshal ( bb , ve jObj )